sql-plugin/src/main/scala/com/nvidia/spark/rapids/DateUtils.scala

```diff
@@ -57,7 +59,13 @@ object DateUtils {
   val YESTERDAY = "yesterday"
   val TOMORROW = "tomorrow"
 
-  def specialDatesDays: Map[String, Int] = {
+  private lazy val isSpark320OrLater: Boolean = {
+    ShimLoader.getSparkShims.getSparkShimVersion.toString >= "3.2"
```

Comment:
My main concern with doing it this was is that we are starting to be inventive about how we handle different situations now that we have the shim layer.  I am fine with this here, but at some point we are going to need to think a lot about the pros and cons of each approach so it is more maintainable.  Before the ShimLayer was a single class so when we had to update code for a version or add a new version there was only one relatively small location we had to look at and think about.  Now it is spread all through the shim jars like before, in the sql plugin in the shim v2 directories (still isolated, but a lot bigger and harder to follow in some cases), and with this it starts to be anywhere in the code.  This is what makes me somewhat concerned.  In order to save a few lines of code we are making it much harder to know what the differences are from one version to another.

To ameliorate that could we have a single VersionUtil or something similar would be a central point for dealing with this type of version checks so we can then trace back everywhere that it is used?

Reply:
I agree that we need to use some well-thought-out version comparison methods for this kind of situation. Comparing based on strings also isn't safe. The above conditional would fail with spark version 3.10.0 for example (unlikely situation but you never know).

Another option here is to move the special dates map in to the shim layer, with 320 returning empty maps.

Reply:
Perhaps we can migrate version utils in spark-test to form a general propose VersionUtil?